applications/vessel-history/src/features/settings/components/Encounters.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useCallback } from 'react'
+import cx from 'classnames'
+import { MultiSelect, InputText, IconButton } from '@globalfishingwatch/ui-components'
+import '@globalfishingwatch/ui-components/dist/base.css'
+import { SelectOption } from '@globalfishingwatch/ui-components/dist/select'
+import { MultiSelectOption } from '@globalfishingwatch//ui-components/dist/multi-select'
+import { t } from 'features/i18n/i18n'
+import { SettingsEvents } from '../settings.slice'
+import { useApplySettingsConnect } from '../settings.hooks'
+import styles from './SettingsComponents.module.css'
+
+interface SettingsProps {
+  settings: SettingsEvents
+  section: string
+}
+
+const Encounters: React.FC<SettingsProps> = (props): React.ReactElement => {
+  const { settings, section } = props
+
+  const { setSettingOptions, setSetting } = useApplySettingsConnect()
+
+  const EEZ_OPTIONS: SelectOption[] = [
```

Comment:
We should get or build this list dynamically.

Reply:
we know that, when I know where to take the list I will do, also I don't know if the different settings use the same EEZ, so I created 3 different list por the moment

Reply:
I doubt they would differ for different event types. Beyond that we could define where in our state we are planning on storing all these lists using these values as initialState and adding a comment to be completed/implemented later, with that we can define the selector to retrieve it and also use it here.